google-cloud-tools-plugin/ultimate/testSrc/com/google/cloud/tools/intellij/appengine/facet/impl/AppEngineStandardUltimateWebIntegrationTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.facet.impl;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.intellij.facet.FacetManager;
+import com.intellij.javaee.web.WebRoot;
+import com.intellij.javaee.web.facet.WebFacet;
+import com.intellij.openapi.application.Result;
+import com.intellij.openapi.application.WriteAction;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.roots.ModifiableRootModel;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.testFramework.PlatformTestCase;
+import com.intellij.util.descriptors.ConfigFile;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import org.jetbrains.annotations.NotNull;
+
+/** Tests for {@link AppEngineStandardUltimateWebIntegration}. */
+public class AppEngineStandardUltimateWebIntegrationTest extends PlatformTestCase {
```

Comment:
Could we probably think about using `CloudToolsRule` and `@Mock/@TestService` here (it could be a bit cleaner to read and in line with the latest tests), or this test requires using `PlatformTestCase`?